junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -286,10 +287,16 @@ static class CloseablePath implements CloseableResource {
 
 		private CloseablePath(TempDirFactory factory, CleanupMode cleanupMode, AnnotatedElementContext elementContext,
 				ExtensionContext extensionContext) throws Exception {
-			this.dir = validateTempDirectory(factory.createTempDirectory(elementContext, extensionContext));
-			this.factory = factory;
-			this.cleanupMode = cleanupMode;
-			this.extensionContext = extensionContext;
+			try {
+				this.dir = validateTempDirectory(factory.createTempDirectory(elementContext, extensionContext));
+				this.factory = factory;
+				this.cleanupMode = cleanupMode;
+				this.extensionContext = extensionContext;
+			}
+			catch (PreconditionViolationException e) {
+				factory.close();
+				throw e;
```

Comment:
The `Path` returned by `createTempDirectory` should be deleted in case of validation errors